presto-main/src/main/java/com/facebook/presto/sql/relational/SqlToRowExpressionTranslator.java

```diff
@@ -787,6 +787,19 @@ protected RowExpression visitSubscriptExpression(SubscriptExpression node, Void
             RowExpression base = process(node.getBase(), context);
             RowExpression index = process(node.getIndex(), context);
 
+            // this block will handle row subscript, converts the ROW_CONSTRUCTOR with subscript to a DEREFERENCE expression
+            if (base.getType() instanceof RowType) {
+                checkState(index instanceof ConstantExpression, "Subscript expression on ROW requires a ConstantExpression");
+                ConstantExpression pos = (ConstantExpression) index;
+                checkState(pos.getValue() instanceof Long, "ConstantExpression should contain a valid integer index into the row");
+                Long offset = (Long) pos.getValue();
```

Comment:
Check if this value is null.

Reply:
the checkState will fail anyways. so I don't think we need null check